index.js

```diff
@@ -42,107 +47,63 @@ class MiHumidifier {
 
         // Current state
-        this.humidifierService
-            .getCharacteristic(Characteristic.CurrentHumidifierDehumidifierState)
-            .setProps({
-                validValues: [0, 2]
-            })
-            .on('get', this.getCurrentHumidifierState.bind(this));
+        this.registerCharacteristic(CharacteristicOperation.GET, this.humidifierService, Characteristic.CurrentHumidifierDehumidifierState, true, this.getCurrentHumidifierState, { validValues: [0, 2] });
 
         // Target state (only humidifier is supported)
         this.humidifierService
             .getCharacteristic(Characteristic.TargetHumidifierDehumidifierState)
-            .setProps({
-                validValues: [1]
-            })
+            .setProps({ validValues: [1] })
             .setValue(Characteristic.TargetHumidifierDehumidifierState.HUMIDIFIER);
 
         // Current relative humidity
-        if (this.humidifier.humidityGetName) {
-            this.humidifierService
-                .getCharacteristic(Characteristic.CurrentRelativeHumidity)
-                .on('get', this.getCurrentRelativeHumidity.bind(this));
-        }
+        this.registerCharacteristic(CharacteristicOperation.GET, this.humidifierService, Characteristic.CurrentRelativeHumidity, this.humidifier.humidityGetName, this.getCurrentRelativeHumidity);
 
         // Target relative humidity
-        if (this.humidifier.targetHumidityGetName && this.humidifier.targetHumiditySetName) {
-            this.humidifierService
-                .getCharacteristic(Characteristic.RelativeHumidityHumidifierThreshold)
-                .on('get', this.getTargetRelativeHumidity.bind(this))
-                .on('set', this.setTargetRelativeHumidity.bind(this));
-        }
+        this.registerCharacteristic(CharacteristicOperation.GET, this.humidifierService, Characteristic.RelativeHumidityHumidifierThreshold, this.humidifier.targetHumidityGetName, this.getTargetRelativeHumidity);
+        this.registerCharacteristic(CharacteristicOperation.SET, this.humidifierService, Characteristic.RelativeHumidityHumidifierThreshold, this.humidifier.targetHumiditySetName, this.setTargetRelativeHumidity);
 
         // Current water level (remaining water level)
-        if (this.humidifier.waterLevelGetName) {
-            this.humidifierService
-                .getCharacteristic(Characteristic.WaterLevel)
-                .setProps({
-                    minValue: this.humidifier.waterLevelLimits.Min,
-                    maxValue: this.humidifier.waterLevelLimits.Max
-                })
-                .on('get', this.getWaterLevel.bind(this));
-        }
+        this.registerCharacteristic(CharacteristicOperation.GET, this.humidifierService, Characteristic.WaterLevel, this.humidifier.waterLevelGetName, this.getWaterLevel, {
+            minValue: this.humidifier.waterLevelLimits.Min,
+            maxValue: this.humidifier.waterLevelLimits.Max
+        });
```

Comment:
Water level is not supported on `v1` so `this.humidifier.waterLevelLimits` will be `undefined` and plugin will crash here
```suggestion
        if (this.humidifier.waterLevelGetName) {
            this.registerCharacteristic(CharacteristicOperation.GET, this.humidifierService, Characteristic.WaterLevel, this.humidifier.waterLevelGetName, this.getWaterLevel, {
                minValue: this.humidifier.waterLevelLimits.Min,
                maxValue: this.humidifier.waterLevelLimits.Max
            });
        }
```

Reply:
Fixed